inc/page_cache/namespace.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Altis\Cloud\Page_Cache;
+
+use function Altis\Cloud\get_config;
+
+function bootstrap() {
+	global $batcache;
+	$batcache = [];
+
+	// Ignore common query params.
+	ignore_query_params();
+
+	// Forward CloudFront cache headers.
+	handle_cloudfront_headers();
+}
+
+function ignore_query_params() {
+	global $batcache;
+	$batcache = $batcache ?? [];
+
+	$config = get_config();
+	$ignored_query_string_params = $config['cache']['ignored-query-string-params'] ?? [
+		'utm_campaign',
+		'utm_medium',
+		'utm_source',
+		'utm_content',
+		'fbclid',
+		'_ga',
+	];
+
+	$batcache['ignored_query_string_params'] = $ignored_query_string_params;
+}
+
+function handle_cloudfront_headers() {
+	global $batcache;
+	$batcache = $batcache ?? [];
+
+	$config = get_config();
+	$unique_headers = $config['cache']['unique-headers'] ?? [
+		'CloudFront-Is-Desktop-Viewer',
+		'CloudFront-Is-Mobile-Viewer',
+		'CloudFront-Is-SmartTV-Viewer',
+		'CloudFront-Is-Tablet-Viewer',
+		'CloudFront-Viewer-Country',
```

Comment:
I think by default we don't want to make country (or maybe any?) of these part of the `unique` key? For example, I think you wanted this for analytics, whereby we wanted to echo back the `CloudFront-Viewer-Country` _pre_ batcache, i.e. not have a page cache per country.

Reply:
Right I see, was thinking that was tied to the page cache too for some reason but that's just at the CloudFront level. So those headers can be reflected back regardless if they're set - the config can be used to opt them into the `unique` key.

This will require an infrastructure change too so I'll raise an issue for that before we go any further here.